### DISCUSSION TITLE
How to set the mirth jdk version

### DISCUSSION THREAD

**Discussion by ourlang** [nextgenhealthcare/connect#5908] [Q&A]
There are multiple versions of jdk on the computer, such as jdk8 and jdk17. How to set which jdk to use for mirth

**Comment by pacmano1** [comment]
@ourlang , stop posing in issues please, post to discussions unless you have a confirmed bug or feature reqeust.

**Reply by pacmano1** [reply]
I moved your question to discussions.

**Reply by pacmano1** [reply]
As far as your question is concerned, you need to post:

1) the operating system
2) how you installed java
3) What "mirth" you are talking about, the admin tool or the server.

**Reply by ourlang** [reply]
OS: Windows 10
cpu:11th Gen Intel(R) Core(TM) i5-1135G7 @ 2.40GHz   2.42 GHz
RAW:40G
 
JDK1.8 Using the Windows installation version

Attempting to install a higher version of JDK due to excessive CPU usage temporarily

mirth Serveice(  mcservice.exe)How to set the mirth jdk version  。Now I want to switch to jdk17

**Comment by rjkroll** [comment]
Mirth either uses the bundled version of Java it came with or the version of Java specified in your Windows JAVA_HOME enviromental variables.  

I also found this, not sure if it works:  

```
"If you want to use a non-default java JDK you can do so by creating a file “pref_jre.cfg” in folder “/opt/Mirth Connect/.install4j”

The file should contain a single line with the path to your java home folder; e.g.:

/usr/lib/jvm/java-11-amazon-corretto"
```
